generate_linux_tests.rb

```diff
@@ -0,0 +1,248 @@
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in all
+//  copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//  SOFTWARE.
+//
+//
+//
+//  NOTE: This file was generated by generate_linux_tests.rb
+//
+//  Do NOT edit this file directly as it will be regenerated automatically when needed.
+//
+
+import XCTest
+  eos
+
+  string
+  .sub('<FileName>', File.basename(file_name))
+  .sub('<Date>', Time.now.to_s)
+end
+
+def create_extension_file(file_name, classes)
+  extension_file = file_name.sub! '.swift', '+XCTest.swift'
+  print 'Creating file: ' + extension_file + "\n"
+
+  File.open(extension_file, 'w') do |file|
+    file.write header(extension_file)
+    file.write "\n"
+
+    for class_array in classes
+      file.write 'internal extension ' + class_array[0] + " {\n"
+      file.write '    static var allTests: [(String, (' + class_array[0] + ") -> () throws -> Void)] {\n"
+      file.write "        return [\n"
+
+      class_count = class_array[1].size
+      count = 0
+      for func_name in class_array[1]
+        count += 1
+
+        file.write '            ("' + func_name + '", ' + func_name + (count == class_count ? ")\n" : "),\n")
+      end
+
+      file.write "        ]\n"
+      file.write "    }\n"
+      file.write "}\n"
+    end
+  end
+end
+
+def create_linux_main(tests_directory, all_test_sub_directories, files)
+  file_name = tests_directory + '/LinuxMain.swift'
+  print 'Creating file: ' + file_name + "\n"
+
+  File.open(file_name, 'w') do |file|
+    file.write header(file_name)
+    file.write "\n"
+
+    file.write "#if os(Linux) || os(FreeBSD)\n"
+    for test_sub_directory in all_test_sub_directories.sort { |x, y| x <=> y }
+      file.write '    @testable import ' + test_sub_directory + "\n"
+    end
+    file.write "\n"
+    file.write "    XCTMain(\n        [\n"
+
+    test_cases = []
+    for classes in files
+      for class_array in classes
+        test_cases << class_array[0]
+      end
+    end
+
+    cases_count = test_cases.size
+    count = 0
+    for test_case in test_cases.sort { |x, y| x <=> y }
+      count += 1
+
+      file.write '            testCase(' + test_case + (count == cases_count ? ".allTests)\n" : ".allTests),\n")
+    end
+    file.write "        ]\n    )\n"
+    file.write "#endif\n"
+  end
+end
+
+def parse_source_file(file_name)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Cyclomatic complexity for parse_source_file is too high. [8/6]](https://app.codacy.com/app/FabrizioBrancati/SwiftyBot/pullRequest?prid=3796373)